src/clients/java/src/test/java/com/tigerbeetle/IntegrationTest.java

```diff
@@ -1366,6 +1366,41 @@ public void testAsyncTasks() throws Throwable {
         }
     }
 
+    /**
+     * This test asserts that the client can handle thread interruption.
+     */
+    @Test
+    public void testConcurrentInterruptedTasks() throws Throwable {
+        final int TASKS_COUNT = 256;
+        final var barrier = new CountDownLatch(TASKS_COUNT);
+        final var zeroedId = UInt128.asBytes(0L);
+
+        // Connect to an invalid cluster, so the calls never complete.
+        try (final var client = new Client(clusterId, new String[] {"0"})) {
```

Comment:
Could you also manually smoke-test this with a real cluster, to make sure that interruption is correctly handled at any stage of request lifecycle? I don't think we  need a test here (though it won't harm), but manual smoke test feels in order!